custom_components/frigate/camera.py

```diff
@@ -153,24 +154,28 @@ def __init__(
             self._cam_name
             in self._frigate_config.get("go2rtc", {}).get("streams", {}).keys()
         ):
-            self._restream_type = "rtsp"
-            streaming_template = config_entry.options.get(
-                CONF_RTSP_URL_TEMPLATE, ""
-            ).strip()
-
-            if streaming_template:
-                # Can't use homeassistant.helpers.template as it requires hass which
-                # is not available in the constructor, so use direct jinja2
-                # template instead. This means templates cannot access HomeAssistant
-                # state, but rather only the camera config.
-                self._stream_source = Template(streaming_template).render(
-                    **self._camera_config
-                )
+            if config_entry.options.get(CONF_ENABLE_WEBRTC, False):
+                self._restream_type = "webrtc"
+                self._attr_frontend_stream_type = StreamType.WEB_RTC
             else:
-                self._stream_source = (
-                    f"rtsp://{URL(self._url).host}:8554/{self._cam_name}"
-                )
-
+                self._restream_type = "rtsp"
+                self._attr_frontend_stream_type = StreamType.HLS
+                streaming_template = config_entry.options.get(
+                    CONF_RTSP_URL_TEMPLATE, ""
+                ).strip()
+
+                if streaming_template:
+                    # Can't use homeassistant.helpers.template as it requires hass which
```

Comment:
If we want to use HA templates we can move this code into `async_added_to_hass()`

Reply:
Honestly I think it's fine as-is, at least no-one has asked for this.